src/dirent.rs

```diff
@@ -0,0 +1,40 @@
+use {Result, Error, Errno};
+use libc::{self, DIR};
+use std::os::unix::io::RawFd;
+use errno;
+
+pub struct Dir {
+    pub handle: *mut DIR,
```

Comment:
Sorry, you're right. `Into<*mut DIR>` is what you'd want to do.

There is also `dirfd` which you could use to implement the `*RawFd` traits if you wanted to.